<?xml version="1.0"?>
<launch>
    <rosparam command="load" file="$(env HOME)/Git/Masterprojekt/src/run_basics/launch/params.yaml" />


    <!-- run cropper for pintcloud data to match size of images -->
    <node pkg="run_basics" type="Synchronizer" name="Synchronizer" output="screen"
          args="" />

    <!-- run imager republisher for converting compressed to raw -->
    <node pkg="image_transport" type="republish" name="pup_imgraw"
          args="compressed in:=/BugaSegm/synchronized/usb_cam/image_raw raw out:=/BugaSegm/img_raw" />

    <!-- run cropper for pintcloud data to match size of images -->
    <node pkg="run_basics" type="PcPreprocess" name="PcPreprocess" output="screen"
          args="" />

    <!-- run segm2object to create image with objects and classes for SegNet image -->
    <node pkg="SegNet" type="SegNet.py" name="SegNet" output="screen"
          args="--model $(env HOME)/Git/Masterprojekt/src/SegNet/Example_Models/segnet_model_driving_webdemo.prototxt
                --weights $(env HOME)/Git/Masterprojekt/src/SegNet/Example_Models/segnet_weights_driving_webdemo.caffemodel
                --colours $(env HOME)/Git/Masterprojekt/src/SegNet/src/camvid12.png
                --caffe $(env HOME)/caffe-segnet/" />

    <!-- run segm2object to create image with objects and classes for SegNet image -->
    <node pkg="run_basics" type="Combinator" name="Combinator" output="screen"
          args="" />

    <!-- run ObjectCalculator to calculate Object-information from the PointCloud -->
    <node pkg="run_basics" type="ObjectCalculator.py" name="ObjectCalculator" output="screen"
          args="" />

    <!-- run final_visualizer to show the results -->
    <node pkg="run_basics" type="FinalVisualizer" name="FinalVisualizer" output="screen"
          args="" />

    <!-- run rviz -->
    <!--
    <node pkg="rviz" type="rviz" name="rviz_gui"
          args="-d $(env HOME)/Masterprojekt/src/run_basics/rviz/basics.rviz" />
    -->
</launch>
